manifests/config.pp

```diff
@@ -45,6 +45,7 @@
   }
 
   $min_puma_threads = pick($foreman::foreman_service_puma_threads_min, $foreman::foreman_service_puma_threads_max)
+  $puma_workers = pick($foreman::foreman_service_puma_workers, floor(min($facts['processors']['count'] * 1.5, (($facts['memory']['system']['total_bytes']/(1024 * 1024 * 1024)) - 1.5))))
```

Comment:
```suggestion
  # CPU based calculation is based on https://github.com/puma/puma/blob/master/docs/deployment.md#mri
  $default_puma_workers_cpu_limit = floor($facts['processors']['count'] * 1.5)
  # Memory based calculation is based on https://docs.gitlab.com/ee/install/requirements.html#puma-settings
  $default_puma_workers_memory_limit = floor($facts['memory']['system']['total_bytes']/(1024 * 1024 * 1024) - 1.5)
  # The default is the min of these two limits. It is not saved as a parameter, to allow it to be more dynamic in case
  # of hardware resource changes
  $default_puma_workers = min($default_puma_workers_cpu_limit, $default_puma_workers_memory_limit)
  # It can still be overridden if the user desires to set it to a fixed value
  $puma_workers = pick($foreman::foreman_service_puma_workers, $default_puma_workers)
```

Trying to break this up a bit for readability, and add comments explaining why it is structured this way

Reply:
```suggestion
  # CPU based calculation is based on https://github.com/puma/puma/blob/master/docs/deployment.md#mri
  # Memory based calculation is based on https://docs.gitlab.com/ee/install/requirements.html#puma-settings
  # The default is the min of these two limits. It is not saved as a parameter, to allow it to be more dynamic in case
  # of hardware resource changes
  $default_puma_workers_cpu_limit = floor($facts['processors']['count'] * 1.5)
  $default_puma_workers_memory_limit = floor($facts['memory']['system']['total_bytes']/(1024 * 1024 * 1024) - 1.5)
  $default_puma_workers = min($default_puma_workers_cpu_limit, $default_puma_workers_memory_limit)

  # The default can still be overridden if the user desires to set it to a fixed value
  $puma_workers = pick($foreman::foreman_service_puma_workers, $default_puma_workers)
```

Reply:
I'm not sure you need additional variables. Alternative:
```suggestion
  $puma_workers = pick(
    $foreman::foreman_service_puma_workers,
    floor(min(
      $facts['processors']['count'] * 1.5,
      ($facts['memory']['system']['total_bytes']/(1024 * 1024 * 1024)) - 1.5
     ))
   )
```

Reply:
I did a hybrid of these suggestions, adding some inline documentation, updating the param documentation and updating the style of the code.